src/components/molecules/Chatbox/Chatbox.tsx

```diff
@@ -71,6 +80,19 @@ export const Chatbox: React.FC<ChatboxProps> = ({
     reset();
   });
 
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false);
+
+  const addEmoji = (emoji: CustomEmoji) => {
+    if (emoji.colons) {
+      const message = getValues("message");
+      setValue("message", message + emoji.colons);
+    }
+    setShowEmojiPicker(false);
+  };
+  const toggleEmojiPicker = () => {
+    setShowEmojiPicker(!showEmojiPicker);
+  };
```

Comment:
```suggestion
  const {
    isShown: isEmojiPickerVisible,
    hide: hideEmojiPicker,
    toggle: toggleEmojiPicker,
  } = useShowHide();

  const addEmoji = (emoji: CustomEmoji) => {
    if (emoji.colons) {
      const message = getValues("message");
      setValue("message", message + emoji.colons);
    }
    
    hideEmojiPicker();
  };
```

Reply:
`addEmoji` should be `useCallback`'d as well

Reply:
What is the intent of the logic in `addEmoji` here? Particularly the `emoji.colons` check + `getValues` call, etc? Can `CustomEmoji` ever not have `.colons` set?

Reply:
FWIW: I no longer see `addEmoji` here 

<details>
<summary>only in ChatMessageBox.tsx and as a useCallback</summary> 

```shell
(git)lena:~/proj/misc/sparkle/sparkle[feature/markdown]git
$> git grep -2 addEmoj
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-  } = useShowHide();
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx:  const addEmoji = useCallback(
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-    (emoji: BaseEmoji) => {
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-      if (emoji.native) {
--
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-      {isEmojiPickerVisible && (
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-        <div className="Chatbox__emoji-picker">
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx:          <Picker theme={"dark"} onSelect={addEmoji} native />
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-        </div>
src/components/molecules/ChatMessageBox/ChatMessageBox.tsx-      )}

```
</details>
and also no `.colons` there ... so may be it was already resolved by @margulies or @soichih

Reply:
It was replaced with `emoji.native` check, which basically means that only native(standard icons can be used). Also, this part was added, which ensures that users see only native icons

![image](https://user-images.githubusercontent.com/63194656/120830775-85154880-c567-11eb-9296-2247eeb815e7.png)